app/services/profile_fields/remove.rb

```diff
@@ -16,8 +16,10 @@ def initialize(id)
     def call
       @profile_field = ProfileField.find(@id)
       if @profile_field.destroy
-        accessor = profile_field.attribute_name.to_s
-        Profile.undef_method(accessor) if accessor.in?(Profile.attributes)
+        accessor = @profile_field.attribute_name.to_sym
+        store_attributes = Profile.stored_attributes.fetch(:data) { [] }
+        Profile.undef_method(accessor) if accessor.in?(store_attributes)
```

Comment:
Should we also undef the setter `"#{attribute_name}=".to_sym`?

Reply:
Done. ✅